See https://trac.macports.org/ticket/61910
V currently only supports dynamically setting C Flags from environment variables.

--- vlib/pg/pg.v.original	2022-04-23 18:59:01.000000000 +0100
+++ vlib/pg/pg.v	2022-04-23 20:07:41.000000000 +0100
@@ -4,9 +4,8 @@
 
 #flag -lpq
 #flag linux -I/usr/include/postgresql
-#flag darwin -I/opt/local/include/postgresql11
-#flag darwin -I/opt/homebrew/include
-#flag darwin -L/opt/homebrew/lib
+#flag darwin -I@PREFIX@/include/postgresql$env('POSTGRESQL_VERSION')
+#flag darwin -L@PREFIX@/lib/postgresql$env('POSTGRESQL_VERSION')
 #flag windows -I @VEXEROOT/thirdparty/pg/include
 #flag windows -L @VEXEROOT/thirdparty/pg/win64
 
